larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -37,30 +37,30 @@ const int kTensorNotAllocated = -1;
 
 typedef struct {
   // input tensor dimensions
-  std::int64_t batch{0};
-  std::int64_t input_width{0};
-  std::int64_t input_height{0};
+  std::int32_t batch{0};
+  std::int32_t input_width{0};
+  std::int32_t input_height{0};
 
   // filters tensor dimensions
-  std::int64_t filter_width{0};
-  std::int64_t filter_height{0};
-  std::int64_t channels_in{0};
-  std::int64_t channels_out{0};
+  std::int32_t filter_width{0};
+  std::int32_t filter_height{0};
+  std::int32_t channels_in{0};
+  std::int32_t channels_out{0};
 
   // strides
-  std::int64_t strides[4] = {};
+  std::int32_t strides[4] = {};
 
   // dilations
-  std::int64_t dilations[4] = {};
+  std::int32_t dilations[4] = {};
 
   // padding
   TfLitePadding padding_type{};
   TfLitePaddingValues padding_values{};
   int pad_value = 0;  // Must be 0 or 1
 
   // output tensor dimensions
-  std::int64_t out_width{0};
-  std::int64_t out_height{0};
+  std::int32_t out_width{0};
+  std::int32_t out_height{0};
```

Comment:
I now can't remember why I had to change these to int32 rather than int64 (it was to fix some sort of compilation error), but I can't see any issue with doing so? I don't think we need a double-word to store input/filter/output dimensions.

Reply:
I think the `int64` was a remnant of some flatbuffer parsing logic where it seemed like a flatbuffer `int8`/`int32`/`int64` could alwasy be read by a `.GetAsInt64()` but not the other way around, so we initially went with int64 to be sure that it worked. Now that we know how to change the flatbuffer types (using the mlir IR) we can go back to int32 (or even int16 since image sizes will probably never get that big).
I think at some point in the future we want to follow the tflite implementation, where we also change the `int64 strides[4]` and `int64 dilations[4]` to `int32 stride_height, stride_width, dilation_height, dilation_width`, i.e. remove the useless batch and channel strides. When we do that we can also change all the int64 to int32. For now we can leave it.

Reply:
Makes sense, and apparently it works fine with `uint64` so I'm not sure what the previous problem I had was. Have reverted in 7ff1f0b.